src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -100,6 +103,86 @@ def pileupSizeTask(self):
             self.logger.exception(msg)
         return cmsDict, cusDict
 
+    def partialPileupTask(self, rseExpression):
+        """
+        Execute partial pileup placement according to the following logic:
+
+        - get rucio DIDs (datasets, in CMS name blocks) for our pileup document
+        - get portion of DIDs based on ceil(containerFraction * num_rucio_datasets)
+        - create new container DID in Rucio via Rucio wrapper createContainer API
+          which calls rucio CLI add_container
+        - update MSPileup JSON with
+          - create new custom Name as pileup+extention
+          - add new transition record and update MSPileup document
+        - we call attachDIDs Rucio wrapper API with our set of DIDs and rses from pileup document
+        - create new rules for custom DID using either FNAL disk or T2_CERN sites
+        - set expiration date (to be 24h) for already existing ruleIds from pileup document
+        - add new ruleIds to pileup document
+
+        :param rseExpression: string, Rucio rse expression, see
+        https://rucio.cern.ch/documentation/started/concepts/rse_expressions/
+        :return: None
+        """
+        spec = {'containerFraction': {"$gt": 0}}
+        docs = self.mgr.getPileup(spec)
+        for doc in docs:
+            self.logger.info("partialPileupTask process %s", doc)
+            fraction = doc['containerFraction']
+
+            # check previous fraction value and proceed only if has changed
+            lastFraction = 0
+            transition = doc.get('trainsition', [])
+            if transition:
+                lastFraction = transition[-1]['containerFraction']
+            if lastFraction == fraction:
+                # do nothing since there is no transition, i.e. skip this record
+                continue
+
+            # get rucio DIDs (datasets, in CMS name blocks) for our pileup document
+            pname = doc['pileupName']
```

Comment:
I am going to leave a comment for this, but at the main ticket discussion instead of in place.